server/Arcadia.Assistant.Web/Startup.cs

```diff
@@ -75,18 +74,39 @@ public void ConfigureServices(IServiceCollection services)
                         //x.CustomSchemaIds(t => t.FullName);
                     });
 
+            services.AddScoped<AuthenticationEvents>();
+
             services
                 .AddAuthentication(options => { options.DefaultScheme = JwtBearerDefaults.AuthenticationScheme; })
                 .AddJwtBearer(
                     jwtOptions =>
-                        {
-                            jwtOptions.Audience = appSettings.Security.ClientId;
-                            jwtOptions.MetadataAddress = appSettings.Security.OpenIdConfigurationUrl;
-                            jwtOptions.Events = new JwtEventsHandler();
-                        });
+                    {
+                        jwtOptions.Audience = appSettings.Security.ClientId;
+                        jwtOptions.MetadataAddress = appSettings.Security.OpenIdConfigurationUrl;
+                        jwtOptions.Events = new JwtEventsHandler();
+                    })
+                .AddBasicAuthentication(
+                    basicOptions =>
+                    {
+                        basicOptions.Realm = this.AppSettings.ServiceEndpointsAuthentication.Realm;
+                        basicOptions.EventsType = typeof(AuthenticationEvents);
+                    });
 
             services
-                .AddAuthorization(options => options.AddPolicy(Policies.UserIsEmployee, policy => policy.Requirements.Add(new UserIsEmployeeRequirement())));
+                .AddAuthorization(options =>
+                {
+                    options.AddPolicy(Policies.UserIsEmployee, policy =>
+                    {
+                        policy.AddAuthenticationSchemes(JwtBearerDefaults.AuthenticationScheme);
```

Comment:
is this `addauthenticationschemes` required in this policy?

Reply:
No. But as I understood, when `Authorize` attribute is used without policy specified, it used first defined policy matched by authorization scheme or without scheme at all. I decided to specify schemes these policies are used with.